penguin/engine.py

```
@@ -2372,6 +2473,46 @@ async def _handle_responses_tool_calls(
+        tool_call: ToolCall,
+    ) -> None:
+        """Persist one lightweight tool-call record on the current session."""
+
+        try:
+            session = (
+                cm.get_current_session() if hasattr(cm, "get_current_session") else None
+            )
+            add_record = getattr(session, "add_tool_call_record", None)
+            if callable(add_record):
+                add_record(tool_call_record_from_tool_call(tool_call))
+        except Exception as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)